docs/generate-evidence-strings.md

```diff
@@ -0,0 +1,279 @@
+${BSUB_CMDLINE} -K -n 8 -M 16G \
+  -o ${BATCH_ROOT}/logs/convert_clinvar_files.out \
+  -e ${BATCH_ROOT}/logs/convert_clinvar_files.err \
+  java -Xmx15G -jar ${CODE_ROOT}/clinvar-xml-parser/target/clinvar-parser-1.0-SNAPSHOT-jar-with-dependencies.jar \
+  -i ${BATCH_ROOT}/clinvar/ClinVarFullRelease_${CLINVAR_RELEASE}.xml.gz \
+  -o ${BATCH_ROOT}/clinvar && \
+${BSUB_CMDLINE} -K \
+  -o ${BATCH_ROOT}/logs/filter_clinvar_json.out \
+  -e ${BATCH_ROOT}/logs/filter_clinvar_json.err \
+  python3 ${CODE_ROOT}/bin/clinvar_jsons/extract_pathogenic_and_likely_pathogenic_variants.py \
+  -i ${BATCH_ROOT}/clinvar/clinvar.json.gz \
+  -o ${BATCH_ROOT}/clinvar/clinvar.filtered.json.gz && \
+${BSUB_CMDLINE} -K \
+  -o ${BATCH_ROOT}/logs/consequence_repeat_expansion.out \
+  -e ${BATCH_ROOT}/logs/consequence_repeat_expansion.err \
+  python3 ${CODE_ROOT}/vep-mapping-pipeline/run_repeat_expansion_variants.py \
+  --clinvar-summary-tsv ${BATCH_ROOT}/clinvar/variant_summary_${CLINVAR_RELEASE}.txt.gz \
+  --output-consequences ${BATCH_ROOT}/gene_mapping/consequences_1_repeat.tsv \
+  --output-dataframe ${BATCH_ROOT}/gene_mapping/repeat_dataframe.tsv && \
+${BSUB_CMDLINE} -K \
+  -o ${BATCH_ROOT}/logs/consequence_vep.out \
+  -e ${BATCH_ROOT}/logs/consequence_vep.err \
+  bash ${CODE_ROOT}/vep-mapping-pipeline/run_consequence_mapping.sh \
+  ${BATCH_ROOT}/clinvar/clinvar_${CLINVAR_RELEASE_YEAR}${CLINVAR_RELEASE_MONTH}??.vcf.gz \
+  ${BATCH_ROOT}/gene_mapping/consequences_2_vep.tsv && \
+cat \
+  ${BATCH_ROOT}/gene_mapping/consequences_1_repeat.tsv \
+  ${BATCH_ROOT}/gene_mapping/consequences_2_vep.tsv \
+  > ${BATCH_ROOT}/gene_mapping/consequences_3_combined.tsv && \
+${BSUB_CMDLINE} -K \
+  -M 10G \
+  -o ${BATCH_ROOT}/logs/evidence_string_generation.out \
+  -e ${BATCH_ROOT}/logs/evidence_string_generation.err \
+  python3 ${CODE_ROOT}/bin/evidence_string_generation.py \
+  -e ${BATCH_ROOT_BASE}/manual_curation/latest_mappings.tsv \
+  -g ${BATCH_ROOT}/gene_mapping/consequences_3_combined.tsv \
+  -j ${BATCH_ROOT}/clinvar/clinvar.filtered.json.gz \
+  --ot-schema ${BATCH_ROOT}/evidence_strings/opentargets-${OT_SCHEMA_VERSION}.json \
+  --out ${BATCH_ROOT}/evidence_strings/ && \
+${BSUB_CMDLINE} -K \
+  -o ${BATCH_ROOT}/logs/traits_to_zooma_format.out \
+  -e ${BATCH_ROOT}/logs/traits_to_zooma_format.err \
+  python3 ${CODE_ROOT}/bin/clinvar_jsons/traits_to_zooma_format.py \
+  -i ${BATCH_ROOT}/clinvar/clinvar.filtered.json.gz \
+  -o ${BATCH_ROOT}/clinvar/clinvar_xrefs.txt
+```
+
+## 5. Manual follow-up actions
+
+### Validate the evidence strings
+Generated evidence strings must be additionally validated using tool provided by Open Targets:
+```bash
+pip3 install --upgrade pip
+pip3 install --upgrade opentargets-validator==${OT_VALIDATOR_VERSION}
+python3 -m opentargets_validator.cli \
+  --schema https://raw.githubusercontent.com/opentargets/json_schema/${OT_SCHEMA_VERSION}/opentargets.json \
+  < ${BATCH_ROOT}/evidence_strings/evidence_strings.json
```

Comment:
Right... the `opentargets_validator` has been giving me all sorts of trouble actually. This is one of the points in the protocol which we do not actually follow, because:

1. When you try to run `opentargets_validator.cli`, it indeed starts spawning countless processes, which never quite seem to complete whatever they're doing. I think I already tried running it as a submitted job, and it didn't finish in 24 hours, after which I killed it.
2. Everything which is the validator doing is validating evidence strings against a schema using `jsonschema` module. This is exactly what is already being done by our evidence string generation pipeline upon generating each evidence string.

Reply:
Still, I have moved running the validator to the automated actions and rewrote it as a submitted command. Also I changed reading the evidence strings from a pipe to a positional argument, maybe it's where the problem was.

Fixed in https://github.com/EBIvariation/eva-cttv-pipeline/pull/99/commits/db40e093be44c401c9d539c1ef51ef0e887180f1